src/libsystemd/sd-id128/id128-util.c

```diff
@@ -191,3 +191,25 @@ int id128_compare_func(const sd_id128_t *a, const sd_id128_t *b) {
 }
 
 DEFINE_HASH_OPS(id128_hash_ops, sd_id128_t, id128_hash_func, id128_compare_func);
+
+bool id128_bogus(sd_id128_t id) {
+
+        static const sd_id128_t bogus[] = {
+                /* Some obviously bogus UUIDs, that make sense to filter out in some cases, for example in the DMI
+                 * ProductID. Note that this is never going to be comprehensive, and hence we only check for the most
+                 * basic nonsense. */
+                SD_ID128_ARRAY(00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00),
+                SD_ID128_ARRAY(01, 01, 01, 01, 01, 01, 01, 01, 01, 01, 01, 01, 01, 01, 01, 01),
+                SD_ID128_ARRAY(ff, ff, ff, ff, ff, ff, ff, ff, ff, ff, ff, ff, ff, ff, ff, ff),
+                SD_ID128_ARRAY(00, 01, 02, 03, 04, 05, 06, 07, 08, 09, 0a, 0b, 0c, 0d, 0e, 0f),
+                SD_ID128_ARRAY(01, 02, 03, 04, 05, 06, 07, 08, 09, 0a, 0b, 0c, 0d, 0e, 0f, 10),
```

Comment:
I'm against this change. The main issue is that this is the wrong place to do this filtering. logind is just forwarding the data from the kernel, and whether this data is useful can only be determined by the final user.

(For example, if I want to report a problem with hardware to the manufacturer, and they ask me to provide the UUID, **any** value is "correct", but we are just need to accurately report whatever is there. Also, for the uses in DHCP, we want to match what other implementations do, and again, the fact that the contents are bogus doesn't really matter. OTOH, if we want to use this as any unique identifier, we'd probably set a much higher bar and reject any kind of repeating or linear pattern. So doing this superficial test is useless.)

Reply:
hmm, would you be fine if i tone it down, and we only check for null and all bits one?

THing is, there are plenty bioses around that carry no uuid and the way they encode it is by setting the field to all 00 and ff, as that's how people usually mark "does not apply", and such a logic I'd say even makes sense. Hence, I think it would only be smart of us if we'd turn these two cases to an explicit "i don't have a product id" indication.

Reply:
Yeah, `00` and `FF` seems OK.